/*
 * This file was automatically generated by EvoSuite
 * Thu Aug 04 20:58:51 GMT 2016
 */

package com.pearson.eidetic.driver.threads.subthreads;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import com.amazonaws.regions.Region;
import com.amazonaws.services.ec2.model.Tag;
import com.amazonaws.services.ec2.model.Volume;
import com.pearson.eidetic.driver.threads.subthreads.SnapshotVolumeSyncValidator;
import java.util.ArrayList;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true) 
public class SnapshotVolumeSyncValidator_ESTest extends SnapshotVolumeSyncValidator_ESTest_scaffolding {

  @Test
  public void test0()  throws Throwable  {
      ArrayList<Volume> arrayList0 = new ArrayList<Volume>();
      SnapshotVolumeSyncValidator snapshotVolumeSyncValidator0 = new SnapshotVolumeSyncValidator("\", stacktrace=\"", "c)^Rj&F7", "\", stacktrace=\"", (Integer) null, (Integer) null, (Region) null, arrayList0);
      snapshotVolumeSyncValidator0.isFinished();
      // Undeclared exception!
      try { 
        snapshotVolumeSyncValidator0.run();
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("com.pearson.eidetic.driver.threads.subthreads.SnapshotVolumeSyncValidator", e);
      }
  }

  @Test
  public void test1()  throws Throwable  {
      SnapshotVolumeSyncValidator snapshotVolumeSyncValidator0 = new SnapshotVolumeSyncValidator("azk}", "azk}", "", (Integer) null, (Integer) null, (Region) null, (ArrayList<Volume>) null);
      // Undeclared exception!
      try { 
        snapshotVolumeSyncValidator0.connect((Region) null, "9P2)p8ZPk/1@:S/d(M", "9P2)p8ZPk/1@:S/d(M");
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("com.pearson.eidetic.driver.threads.subthreads.SnapshotVolumeSyncValidator", e);
      }
  }

  @Test
  public void test2()  throws Throwable  {
      Volume volume0 = new Volume();
      Tag[] tagArray0 = new Tag[1];
      Tag tag0 = new Tag();
      tag0.withKey("RankNames: ");
      tag0.withValue("RankNames: ");
      tagArray0[0] = tag0;
  }
}
